### PR TITLE
sci-libs/hdf5: drop autotools.eclass, bump EAPI to 6

### PR DESCRIPTION
**Pull request by krigstask**

### CODE REVIEW

File: sci-libs/hdf5/hdf5-1.10.1.ebuild [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=6
 
 FORTRAN_NEEDED=fortran
-AUTOTOOLS_AUTORECONF=1
 
-inherit autotools-utils eutils fortran-2 flag-o-matic toolchain-funcs multilib versionator prefix
+inherit eutils fortran-2 flag-o-matic toolchain-funcs multilib versionator prefix
```
Comment (a17r) [inline sci-libs/hdf5/hdf5-1.10.1.ebuild]: You probably don't need multilib since you bumped to EAPI 6.

**Comment by heroxbd** [comment]
When you have dropped `autotools.eclass`, a direct call of eautoreconf is probably needed.

**Comment by krigstask** [comment]
> When you have dropped `autotools.eclass`, a direct call of eautoreconf is probably needed

I don't recall what I have read exactly ([Michał's post](https://blogs.gentoo.org/mgorny/2015/11/13/the-ultimate-guide-to-eapi-6/) doesn't seem to contain any pointers), but I was sure that explicit `eautoreconf` isn't required.  I've tested my ebuilds, and they seem to work fine.  Are you sure `eautoreconf` is needed?

**Comment by heroxbd** [comment]
> I don't recall what I have read exactly (Michał's post doesn't seem to contain any pointers), but I was sure that explicit eautoreconf isn't required. I've tested my ebuilds, and they seem to work fine. Are you sure eautoreconf is needed?

The `hdf5-1.8.9-static_libgfortran.patch` and `hdf5-1.8.13-no-messing-ldpath.patch` patches Makefile.am, the patches have no meaning without regenerating Makefile.in from Makefile.am by automake.  If they work fine, we should consider dropping the patches.

**Comment by krigstask** [comment]
I've tried this patch:
```diff
diff --git a/sci-libs/hdf5/hdf5-1.10.1.ebuild b/sci-libs/hdf5/hdf5-1.10.1.ebuild
index b2340fc60..eb80469fa 100644
--- a/sci-libs/hdf5/hdf5-1.10.1.ebuild
+++ b/sci-libs/hdf5/hdf5-1.10.1.ebuild
@@ -72,6 +72,7 @@ src_prepare() {
        # enable shared libs by default for h5cc config utility
        sed -i -e "s/SHLIB:-no/SHLIB:-yes/g" tools/src/misc/h5cc.in || die
        hprefixify m4/libtool.m4
        default
+       eautoreconf
 }
 ```
I ran `ebuild hdf5-1.10.1.ebuild clean configure` _before_ and _after_ this patch had been applied, saved both workdirs and did `diff -ur` on them.  Here's the result: https://gist.github.com/krigstask/1c26a9abdf85495630ac767640a19aac

I can see differences only attributed to PID, date/time and pseudorandom stuff.  Do I miss something?
If patches are of no use anymore, I can drop them in another PR.

**Comment by heroxbd** [comment]
> I ran ebuild hdf5-1.10.1.ebuild clean configure before and after this patch had been applied, saved both workdirs and did diff -ur on them. Here's the result: https://gist.github.com/krigstask/1c26a9abdf85495630ac767640a19aac
> 
> I can see differences only attributed to PID, date/time and pseudorandom stuff. Do I miss something?
> If patches are of no use anymore, I can drop them in another PR.

Nice experiment!  What `USE` flag have you tested with?  The Makefile.am patch is for `USE=fortran`.

**Comment by krigstask** [comment]
That might be my mistake, I tested with `USE="-fortran"` indeed.  I'll continue working on it in a couple of weeks (off to NY vacation now).

**Comment by flappyports** [comment]
no activity

**Pull request closed without merging**